File: sort/Comb_sort/java/CombSort.java
import java.util.*;
import java.math.*;

public class CombSort {

  public static void main(String[] args) {
      int[] elements = init();
      sort(elements);
      Arrays.stream(elements).forEach(i -> System.out.print(i + " "));
      System.out.println();
  }

  public static void sort(int[] elements) {
    int gap = elements.length;
    double shrink = 1.3;
    boolean isSorted = false;
    while(!isSorted) {
      gap = (int)Math.floor(gap / shrink);
      if (gap > 1) {
         isSorted = false;
      } else {
         gap = 1;
         isSorted = true;
      }
      for(int i = 0; i + gap < elements.length; i++) {
        if (elements[i] > elements[i + gap]) {
          int tmp = elements[i + gap];
          elements[i + gap] = elements[i];
          elements[i] = tmp;
        }
      }
    }
  }

  public static int[] init() {
    Scanner sc = new Scanner(System.in);
    System.out.println("Enter the number of elements: ");
    int j, n = sc.nextInt();
    int[] elements = new int[n];
    int gap = n / 2;
    System.out.println("Enter elements: ");
    for(int i = 0; i < n; i++)
      elements[i] = sc.nextInt();
    return elements;
  }
}

File: sort/Comb_sort/python/comb_sort.py
import math

def comb_sort(arr):
    gap = len(arr)
    shrink = 1.3
    is_sorted = False
    aux = arr
    while (is_sorted == False):
        gap = math.floor(gap / shrink)
        if gap > 1:
            is_sorted = False
        else:
            gap = 1
            is_sorted = True
        i = 0
        while (i + gap < len(aux)):
            if aux[i] > aux[i + gap]:
                aux[i], aux[i + gap] = aux[i + gap], aux[i]
                is_sorted = False
            i = i + 1
    return aux

def test():
    list1 = [23,2,7,1,89,20,78,13]
    print(list1)
    print(comb_sort(list1))
    assert (sorted(list1) == comb_sort(list1))

if __name__ == '__main__':
    test()
